### PR TITLE
SDA-4457 : Fix support for conditional addon parameters

### PR DESCRIPTION
**Pull request by ciaranRoche**
Fixes : https://issues.redhat.com/browse/SDA-4457

This change : 
- Bumps ocm-sdk to v0.1.197
- Adds `resource` type to addon parameter interactive menu
- Replaces getAddonParameter call to use `addon-inquiries` endpoint

To verify 
- Create ROSA cluster
- Add Managed API Addon
```
rosa install addon managed-api-service --cluster=<<cluster-name>> --addon-managed-api-service=1 --addon-resource-required=true --cidr-range=10.1.0.0/26 -y 
```
- Verify no errors, and addon is created

### CODE REVIEW

File: cmd/install/addon/cmd.go [hunk 1]
```diff
@@ -218,6 +218,11 @@ func run(cmd *cobra.Command, argv []string) {
 				case "string":
 					input.Default = param.DefaultValue()
 					val, err = interactive.GetString(input)
+				case "resource":
+					var resourceVal int
+					input.Default, _ = strconv.Atoi(param.DefaultValue())
+					resourceVal, err = interactive.GetInt(input)
+					val = fmt.Sprintf("%d", resourceVal)
```
Comment (vkareh) [inline cmd/install/addon/cmd.go:225]: This is the same as the `number` case. You can always do something like
```
case "number", "resource":
// etc...
```

**Comment by vkareh** [comment]
/hold since the addon inquiries is not yet deployed in production

Other than that and my comment above, this looks good to me :+1:

**Comment by ciaranRoche** [comment]
@vkareh can we merge this for the release too?

**Comment by vkareh** [comment]
This is deployed in prod now, right?

**Comment by ciaranRoche** [comment]
Yeah, it was pushed yesterday

**Pull request merged**